src/converter/one/convert/DiffFolded.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.convert;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public final class DiffFolded {
+
+    public static DifferentialResult process(ProfileData profile1, ProfileData profile2) {
+        Map<String, String> stackData = new HashMap<>();
+        Map<String, Long> functionDeltas = new HashMap<>();
+        long maxDelta = 1;
+        
+        Map<String, Long> folded1 = profile1.collapsedData;
+        Map<String, Long> folded2 = profile2.collapsedData;
+        
+        long total1 = folded1.values().stream().mapToLong(Long::longValue).sum();
+        long total2 = folded2.values().stream().mapToLong(Long::longValue).sum();
+
+        boolean shouldNormalize = shouldNormalizeCounts(profile1, profile2, total1, total2);
+
+        Set<String> allStacks = new HashSet<>(folded1.keySet());
+        allStacks.addAll(folded2.keySet());
+
+        for (String stack : allStacks) {
+            long count1 = folded1.getOrDefault(stack, 0L);
+            long count2 = folded2.getOrDefault(stack, 0L);
+
+            // Apply normalization only when appropriate
+            if (shouldNormalize && total1 != total2 && total1 > 0) {
```

Comment:
`shouldNormalize` should contain the rest of the condition, to simplify this if to `if (shouldNormalize)`.